### ISSUE TITLE
Half pixel container rounding

### ISSUE THREAD

**Issue by Judder** [scottjehl/picturefill#533]
Not sure if this is a picturefull issue, or a browser based issue, but when the container is a 1/2 pixel the browser and picturefull are using different maths to produce an image.

Example:

Container width = 611.5 pixels
Original image = 960 x 900
Picturefill image = 612 x 573

which is incorrect as 612 / (960/900) = 573.75 which should then round up to 574 pixels

I think the reason picturefill is calculating it incorrectly is that it doesn't do the rounding on the container first.

Basically the container is

611.5 / (960/900) = 573.28 which should then round down to 573 pixels (which it does at the moment)

In responsive grid design this gives 1 pixel gaps on 1/2 size pixel containers when using % spacing

Any thoughts about a quick workaround?

**Comment by baloneysandwiches** [comment]
Thanks, can you post a reduced test case?

**Comment by baloneysandwiches** [comment]
I'm going to go ahead and close this. Please reopen if you're still experiencing this issue. You should be able to get liquid widths to fully fill their containers using plain old CSS `width: 100%;`.

There has lately been a <a href="https://github.com/jquery/jquery/issues/1724">big push in libraries towards subpixel reporting</a>. The consensus is that helps layout more often than it hurts. So if the browser doesn't round, Picturefill doesn't round.
